### PR TITLE
Adding a method to save client credentials. 

### PR DESCRIPTION
**Pull request by mbruzek**
Normalizing the method names. And adding **docstrings** to the methods.

This is only a proposal. I changed the names of some methods in here that would break existing uses of this interface. I feel they are necessary but that is only my opinion. We should not merge this until we discuss the impacts on other consumers.

### CODE REVIEW

File: requires.py [hunk 1]
```diff
@@ -21,23 +21,47 @@ class EtcdClient(RelationBase):
...
-    def ssl_certificates(self):
-        ''' Return a dict with the client key and certificate '''
+    def get_client_credentials(self):
+        ''' Return a dict with the client certificate, ca and key to
+        communicate with etcd using tls. '''
         return {'client_cert': self.get_remote('client_cert'),
                 'client_key': self.get_remote('client_key'),
                 'client_ca': self.get_remote('client_ca')}
+
+    def save_client_credentials(self, key, cert, ca):
```
Comment (lazypower) [inline requires.py:59]: so we call this method to put bits on disk?

like:

save_client_credentials('/etc/ssl/unit_name/cert.pem', '/etc/ssl/unit_name/key.pem', '/etc/ssl/unit_name/ca.pem')
?

File: requires.py [hunk 2]
```diff
@@ -21,23 +23,47 @@ class EtcdClient(RelationBase):
 
     @hook('{requires:etcd}-relation-{joined,changed}')
     def changed(self):
+        ''' Indicate the relation is connected, and if the relation data is
+        set it is also available. '''
         self.set_state('{relation_name}.connected')
+
         if self.connection_string():
```
Comment (lazypower) [inline requires.py]: i think you mean get_connection_string() here

File: requires.py (hunk 1 above)
Comment (mbruzek) [inline requires.py:59]: yes, I have implemented it thusly in kubernetes:

``` python
        # Define where the etcd tls files will be kept.
        etcd_dir = '/etc/ssl/etcd'
        # Create paths to the etcd client ca, key, and cert file locations.
        ca = os.path.join(etcd_dir, 'client-ca.pem')
        key = os.path.join(etcd_dir, 'client-key.pem')
        cert = os.path.join(etcd_dir, 'client-cert.pem')
        # Save the client credentials (in relation data) to the paths provided.
        reldata.save_client_credentials(key, cert, ca)
        # Update the context so the template has the etcd information.
        context.update({'etcd_dir': etcd_dir,
                        'connection_string': connection_string,
                        'etcd-ca': ca,
                        'etcd-key': key,
                        'etcd-cert': cert})
```

File: requires.py (hunk 2 above)
Comment (mbruzek) [inline requires.py]: Good catch. My bad!

**Comment by lazypower** [comment]
LGTM! 👍

**Pull request merged**